pkg/controllers/namespace/namespace_sync_controller.go

```diff
@@ -89,6 +91,10 @@ func (c *Controller) namespaceShouldBeSynced(namespace string) bool {
 		return false
 	}
 
+	if len(c.SkippedPropagatingNamespaces) > 0 {
+		set := sets.NewString(c.SkippedPropagatingNamespaces...)
+		return !set.Has(namespace)
+	}
```

Comment:
It's a little bit of waste to initialize a set each time.
Can we cache the namespaces by a map or set?

Reply:
ok